### PR TITLE
Clean up CMake scripts.

### PR DESCRIPTION
**Pull request by trivialfis**
@chenqin This is a PR  based on your branch and should be merged after merging your PR.  Please review.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -75,25 +79,15 @@ if(RABIT_BUILD_TESTS)
     add_executable(${test} test/${test}.cc)
     target_link_libraries(${test} rabit_mock_static)
     set_target_properties(${test}  PROPERTIES CXX_STANDARD 11 CXX_STANDARD_REQUIRED ON)
-    install(TARGETS ${test} DESTINATION test)
+    install(TARGETS ${test} DESTINATION test) # Why are we installing these??
```
Comment (chenqin) [inline CMakeLists.txt]: per comments, this is due to we have Makefile in test folder where we run integration tests assuming tests stay in same folder.

File: scripts/travis_setup.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    make -j$(nproc)
+    make install
+
+    popd
+
+    sudo apt-get install python3-pip tree
+fi
+
+if [ ${TRAVIS_OS_NAME} == "osx" ]; then
+    brew install python3
```
Comment (chenqin) [inline scripts/travis_setup.sh]: make sure we have weboscket, kubernetes package to run integration tests.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -11,6 +11,5 @@ target_sources(
 target_include_directories(
         allreduce_base
         PUBLIC
-        ${CMAKE_CURRENT_LIST_DIR}/../dmlc-core/include
-        ${CMAKE_CURRENT_LIST_DIR}/../include
-)
\ No newline at end of file
+	${CMAKE_CURRENT_LIST_DIR}/../../dmlc-core/include
```
Comment (chenqin) [inline src/CMakeLists.txt]: if we assume project layout be
/rabit
/dmlc-core
we need to make sure dmlc-core is downloaded and put in place when we only run rabit tests not xgboost tests.

**Comment by chenqin** [comment]
lots of cleaning, thanks a bunch. Some minor comments.

**Comment by trivialfis** [comment]
@chenqin I thought I was gonna carry it out ...  If you want to finish the CMake refactoring then I will stay out of way and wait for you good news. ;-)

**Comment by trivialfis** [comment]
But the general idea is here.  It's just some more minor works to get everything running smoothly on test farm in both stand-alone build and with XGBoost.

**Comment by chenqin** [comment]
I see, looking for way to undo the merge:)

**Comment by trivialfis** [comment]
Nah.  If you made the changes then I will leave it to you.  Feel free to reach out for any difficulty

**Pull request merged**